### PR TITLE
STM32: Convert USB driver to device tree pinctrl

### PR DESCRIPTION
**Pull request by erwango**
This adds support for device tree pinctrl in STM32 USB driver

Makes use of https://github.com/zephyrproject-rtos/hal_stm32/pull/76

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -382,6 +374,15 @@ static int usb_dc_stm32_init(void)
 	usb_dc_stm32_state.pcd.Init.Sof_enable = 1;
 #endif /* CONFIG_USB_DEVICE_SOF */
 
+	LOG_DBG("Pinctrl signals configuration");
+	status = stm32_dt_pinctrl_configure(usb_pinctrl,
+				     ARRAY_SIZE(usb_pinctrl),
+				     (uint32_t)usb_dc_stm32_state.pcd.Instance);
+	if (status < 0) {
+		LOG_ERR("USB pinctrl setup failed (%d)", status);
+	return -EIO;
```
Comment (gmarull) [inline drivers/usb/device/usb_dc_stm32.c:383]: better propagate error code

```suggestion
	return status;
```

**Review by gmarull — COMMENTED** [review]
Just one small nit

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/usb/device/usb_dc_stm32.c]: Fixed, txs

File: drivers/usb/device/usb_dc_stm32.c [hunk 2]
```diff
@@ -382,6 +374,15 @@ static int usb_dc_stm32_init(void)
 	usb_dc_stm32_state.pcd.Init.Sof_enable = 1;
 #endif /* CONFIG_USB_DEVICE_SOF */
 
+	LOG_DBG("Pinctrl signals configuration");
+	status = stm32_dt_pinctrl_configure(usb_pinctrl,
+				     ARRAY_SIZE(usb_pinctrl),
+				     (uint32_t)usb_dc_stm32_state.pcd.Instance);
+	if (status < 0) {
+		LOG_ERR("USB pinctrl setup failed (%d)", status);
+	return status;
```
Comment (gmarull) [inline drivers/usb/device/usb_dc_stm32.c:383]: I think this line is missing 1 indentation

**Review by loicpoulain — APPROVED** [review]
Looks good. maybe add the link of the dep hal_stm32 PR.

**Review by gmarull — APPROVED** [review]

**Comment by galak** [comment]
@erwango - I merged the HAL side changes if you want to update west.yaml

**Review by galak — APPROVED** [review]

**Comment by erwango** [comment]
@galak, done

**Pull request merged**